### ISSUE TITLE
CA1062 false positive when String.IsNullOrEmpty is used for null check

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2369]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

[CA1062: Validate arguments of public methods](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps
1. Enable CA1062
2. Analyze the following code:

```csharp
public static void A(string input)
{
    if (string.IsNullOrEmpty(input))
    {
        throw new ArgumentException("Invalid input", nameof(input));
    }

    B(input); // This line causes a CA1062 violation, even though "input" is null-checked above.
}

private static void B(string input)
{
    var x = input.Length;
}
```

#### Expected behavior
No warning, since `input` is already checked for `null` via `string.IsNullOrEmpty()`.

#### Actual behavior
CA1062 is reported against `B(input)`.

**Comment by matej-darebnik** [comment]
Hi, I would just like to add that string.IsNullOrWhiteSpace() produces the same false positive.

**Comment by RobSiklos** [comment]
@mavasani Any update on this? It's blocking us from switching from legacy FxCop to the Roslyn analyzers.

**Comment by mavasani** [comment]
I am going to look at all CA1062 false positives today.

**Comment by calumsult** [comment]
Thanks for fixing this!

**Comment by mavasani** [comment]
https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19317.5+45537f7f should have the fix

**Comment by RobSiklos** [comment]
Verified fixed in https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19318.2+bc3ca30e.

Thanks!
